tmt/steps/prepare/install.py

```diff
@@ -10,6 +10,267 @@
 COPR_URL = 'https://copr.fedorainfracloud.org/coprs'
 
 
+class InstallBase(tmt.utils.Common):
+    """ Base class for installation implementations """
+
+    def __init__(
+            self,
+            parent,
+            guest,
+            package,
+            pkg_data,
+            pkg_mgr="",
+            plugin='dnf-plugins-core'):
```

Comment:
Are `pkg_mgr` and `plugin` even optional? It seems epel installation treats it as mandatory input, I would just drop the default value and leave it to subclasses to provide the correct plugin name. Same for `pkgr_mgr`, empty value doesn't look right to me since each and every subclass will have its own `pkg_mgr` anyway.